packages/marshal/src/marshal.js

```diff
@@ -176,68 +256,111 @@ export const makeMarshal = (
       const val = convertSlotToVal(slot, iface);
       valMap.set(index, val);
       return val;
-    }
+    };
 
-    const decodeErrorFromCapData = rawTree => {
-      // Must decode `cause` and `errors` properties
-      const { name, message, errorId } = rawTree;
-      assert.typeof(
-        name,
-        'string',
-        X`invalid error name typeof ${q(typeof name)}`,
-      );
-      assert.typeof(
-        message,
-        'string',
-        X`invalid error message typeof ${q(typeof message)}`,
-      );
-      const EC = getErrorConstructor(`${name}`) || Error;
+    /**
+     * @param {{errorId?: string, message: string, name: string}} errData
+     * @param {(e: unknown) => Passable} decodeRecur
+     * @returns {Error}
+     */
+    const decodeErrorCommon = (errData, decodeRecur) => {
```

Comment:
Should `decodeRecur` have a default value so `decodeErrorFromCapData` can ignore its second argument like `decodeRemotableOrPromiseFromCapData` does?
```suggestion
     * @param {{errorId?: string, message: string, name: string}} errData
     * @param {(e: unknown) => Passable} [decodeRecur]
     * @returns {Error}
     */
    const decodeErrorCommon = (errData, decodeRecur = val => val) => {
```

Reply:
No. Slots are low enough level that not passing the *recur on to the common function is fine. Error handling should not encourage that assumption in too many places. I'd rather make the callers of the error function pass the *recur function explicitly. This also prepares us better to extend the error common functions to handle `cause` and `errors`.

Resolving.